online_editor/views.py

```diff
@@ -16,19 +16,46 @@ def index(request):
     context = {}
     return render(request, 'index.html', context)
 
-def execute(request):
-    code = request.body.decode() + "\n" #decoding byte to code
-    
+def run(request):
+    code = request.body.decode() + "\n"
+
     input_name = settings.INPUT_FILE_PATH
     output_name = settings.OUTPUT_FILE_PATH
     shell_script = settings.RUN_IN_DOCKER_SH_PATH
-    
+    if request.POST['compile_status'] != True:
```

Comment:
**"compile_satatus"** entity is to check whether the code has been executed by back-end before.

 because in our compile step, it will also call docker to execute the code. To reduce the server payload, if the code has been executed, you can search the result in the database, and get the output directly, if the code haven't been executed before, go and execute it in docker. 
So, you can check this entity to see whether it should generate a new code ORM and put the new code into our database then execute the reslut through docker, or just search the already-exist code instance in our database.

Reply:
In the future, after the establish of user module, erery user can view their codind history, thus, this code ORM will be connected to user ORM